hbook/h1d.go

```diff
@@ -284,6 +284,24 @@ func DivH1D(h1, h2 *H1D) *H1D {
 	return hratio
 }
 
+// h1dApply is a helper function to perform bin-by-bin operations on H1D.
+func h1dApply(dst, bins1, bins2 []Bin1D, fct func(b1, b2 Bin1D) (float64, float64)) {
+
+	if len(bins1) != len(bins2) {
```

Comment:
```suggestion
	if len(bins1) != len(bins2) || len(bins1) != len(dst) {
```

(and remove the other `if` statement.